nimbus-core/src/main/java/com/antheminc/oss/nimbus/domain/model/state/internal/DefaultParamState.java

```diff
@@ -928,7 +928,7 @@ private boolean affectToggleActivate(boolean to) {
 	}
 	
 	private boolean isPrimitive() {
-		return ClassUtils.isPrimitiveOrWrapper(getConfig().getReferredClass());
+		return getConfig().getReferredClass().isPrimitive();
```

Comment:
@sandeepmantha ClassUtils.isPrimitiveOrWrapper would have detected both primitives and wrappers vs. Class.isPrimitive is limited. What was the issue that with having the wrapper classes being detected and treated same as primitives?

Reply:
If it gets treated as Object instead of primitive, the test cases have to validate more scenarios such as: frontend interaction for data type (e.g. boolean vs Boolean), json submission to a model which has Primitive Wrapper Object, json submission directly to the Primitive Wrapper. Both the json submission scenarios needs to include null and valid-value test data.

Reply:
Have added test case for primitive boolean and Boolean in both a model and nested model scenario. Have created a scenario in petclinic to verify the usecase.